core/src/datayoga_core/producer.py

```diff
@@ -13,7 +13,7 @@ def __init__(self, msg_id: str, value: Dict[str, Any]):
 class Producer(Block):
 
     @abstractmethod
-    async def produce(self) -> AsyncGenerator[List[Message], None]:
+    def produce(self) -> AsyncGenerator[List[Message], None]:
```

Comment:
In this case, doesn't it mean that all the methods that implement this abstract method should have the same signature?

for example,
    async def produce(self) -> AsyncGenerator[List[Message], None]:

Method 'produce' was expected to be 'non-async', found it instead as 'async'Pylint[W0236:invalid-overridden-method](https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/invalid-overridden-method.html)

Other references:
![image](https://github.com/datayoga-io/datayoga/assets/67013050/6ba37d67-ffe5-4f9b-b8f6-1ec42098ce8f)

Reply:
Reverted.